libs/admin/src/lib/state/admin.effects.ts

```diff
@@ -0,0 +1,118 @@
+import { inject, Injectable } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { fromAppShared } from '@asset-sg/client-shared';
+import { isDecodeError } from '@asset-sg/core';
+import * as RD from '@devexperts/remote-data-ts';
+import { UntilDestroy } from '@ngneat/until-destroy';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { Store } from '@ngrx/store';
+import * as D from 'io-ts/Decoder';
+import { filter, map, merge, switchMap, tap } from 'rxjs';
+
+import { AdminService, User, Workgroup } from '../services/admin.service';
+import * as actions from './admin.actions';
+import { AdminState, AppStateWithAdmin } from './admin.reducer';
+
+@UntilDestroy()
+@Injectable()
+export class AssetAdminEffects {
+  private readonly store = inject(Store<AdminState>);
+  private readonly actions$ = inject(Actions);
+  private readonly adminService = inject(AdminService);
+  private readonly adminStore = inject(Store<AppStateWithAdmin>);
+  private readonly router = inject(Router);
+  private readonly route = inject(ActivatedRoute);
+
+  constructor() {
+    merge(this.store.select(fromAppShared.selectRDReferenceData))
+      .pipe(
+        filter(RD.isFailure),
+        map((e) => e.error),
+        filter(isDecodeError)
+      )
+      .subscribe((e) => {
+        console.error('DecodeError', D.draw(e.cause));
+      });
+  }
+
+  public findUser$ = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(actions.findUser),
+      switchMap(({ userId }) => {
+        return this.adminService.getUserNew(userId).pipe(map((user: any) => actions.setUser({ user })));
+      })
+    );
+  });
+
+  public updateUser$ = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(actions.updateUser),
+      switchMap(({ user }) => {
+        return this.adminService.updateUser(user).pipe(
+          map((user: User) =>
+            actions.setUser({
+              user,
+            })
+          )
+        );
+      })
+    );
+  });
+
+  public findWorkgroup$ = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(actions.findWorkgroup),
+      switchMap(({ workgroupId }) => {
+        return this.adminService
+          .getWorkgroup(workgroupId.toString())
+          .pipe(map((workgroup: Workgroup) => actions.setWorkgroup({ workgroup })));
+      })
+    );
+  });
+
+  public createWorkgroup$ = createEffect(
+    () => {
+      return this.actions$.pipe(
+        ofType(actions.createWorkgroup),
+        switchMap(({ workgroup }) => this.adminService.createWorkgroup(workgroup)),
+        tap((workgroup) => this.router.navigate([`/de/admin/workgroups/${workgroup.id}`], { relativeTo: this.route }))
```

Comment:
Use `switchMap` instead of `tap` so the navigation's promise gets resolved by NgRX.
```suggestion
        switchMap((workgroup) => this.router.navigate([`/de/admin/workgroups/${workgroup.id}`], { relativeTo: this.route }))
```